gobblin-example/src/main/java/org/apache/gobblin/example/hivematerializer/HiveMaterializerSource.java

```diff
@@ -0,0 +1,129 @@
+  public static final String MATERIALIZE_QUERY = HIVE_MATERIALIZER_SOURCE_PREFIX + ".materializeQuery";
+  public static final String OUTPUT_STORAGE_FORMAT = HIVE_MATERIALIZER_SOURCE_PREFIX + ".outputStorageFormat";
+
+  @Override
+  public List<WorkUnit> getWorkunits(SourceState state) {
+    try {
+      FileSystem fs = HadoopUtils.getSourceFileSystem(state);
+      Config config = ConfigUtils.propertiesToConfig(state.getProperties());
+
+      if (state.contains(COPY_TABLE_KEY)) {
+        HiveDataset dataset = getHiveDataset(state.getProp(COPY_TABLE_KEY), fs, state);
+        WorkUnit workUnit = HiveMaterializer.tableCopyWorkUnit(dataset,
+            new StageableTableMetadata(config.getConfig(HIVE_MATERIALIZER_SOURCE_PREFIX), dataset.getTable()), null);
+        HiveTask.disableHiveWatermarker(workUnit);
+        return Lists.newArrayList(workUnit);
+      } else if (state.contains(MATERIALIZE_VIEW)) {
+        HiveDataset dataset = getHiveDataset(state.getProp(MATERIALIZE_VIEW), fs, state);
+        WorkUnit workUnit = HiveMaterializer.viewMaterializationWorkUnit(dataset, getOutputStorageFormat(state),
+            new StageableTableMetadata(config.getConfig(HIVE_MATERIALIZER_SOURCE_PREFIX), dataset.getTable()), null);
+        HiveTask.disableHiveWatermarker(workUnit);
+        return Lists.newArrayList(workUnit);
+      } else if (state.contains(MATERIALIZE_QUERY)) {
+        String query = state.getProp(MATERIALIZE_QUERY);
+        WorkUnit workUnit = HiveMaterializer.queryMaterializationWorkUnit(query, getOutputStorageFormat(state),
+            new StageableTableMetadata(config.getConfig(HIVE_MATERIALIZER_SOURCE_PREFIX), null));
+        HiveTask.disableHiveWatermarker(workUnit);
+        return Lists.newArrayList(workUnit);
+      }
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
+    }
+    throw new RuntimeException(String.format("Must specify either %s, %s, or %s.", COPY_TABLE_KEY, MATERIALIZE_QUERY,
+        MATERIALIZE_VIEW));
+  }
+
+  private HiveDataset getHiveDataset(String tableString, FileSystem fs, State state) throws IOException {
+    try {
+      HiveMetastoreClientPool pool = HiveMetastoreClientPool.get(state.getProperties(),
+          Optional.fromNullable(state.getProp(HIVE_METASTORE_URI_KEY)));
+
+      List<String> tokens = Splitter.on(".").splitToList(tableString);
+      DbAndTable sourceDbAndTable = new DbAndTable(tokens.get(0), tokens.get(1));
+
+      try (AutoReturnableObject<IMetaStoreClient> client = pool.getClient()) {
+        Table sourceTable = new Table(client.get().getTable(sourceDbAndTable.getDb(), sourceDbAndTable.getTable()));
+        return new HiveDataset(fs, pool, sourceTable, ConfigUtils.propertiesToConfig(state.getProperties()));
+      }
+    } catch (TException exc) {
+      throw new RuntimeException(exc);
+    }
+  }
+
+  private HiveConverterUtils.StorageFormat getOutputStorageFormat(State state) {
```

Comment:
This method can be moved to some other places like `HiveMaterializer` as a static method (Just like `HiveMaterializer.viewMaterializationWorkUnit`), as it can be reusable in other places like `LoopingHiveDatasetFinderSource`(Looping coordinator for hive dataset) as well.

Reply:
Not sure about that... the configuration key and default values might be different for different use cases, plus a lot of use cases would just hard code what output format they expect. Ultimately, if refactoring, it would just wrap `HiverConverterUtils.StorageFormat.valueOf`, which is simple enough to use by itself.